apps/condo/domains/user/schema/SignInUserService.js

```
@@ -0,0 +1,221 @@
+ * @param {*} userData
+ * @return {{missingFields: string[], missingRequiredFields: boolean}}
+ */
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (isNil(value)) missingFields.push(field)
+        if (typeof value === 'string' && value.length < 1) missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input SignInUserInput { dv: Int!, sender: SenderFieldInput!, confirmActionToken: ID!, userType: UserTypeType!, userData: SignInUserUserDataInput }',
+        },
+        {
+            access: true,
+            type: 'type SignInUserOutput { user: User, token: ID! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSignInUser,
+            schema: 'signInUser(data: SignInUserInput!): SignInUserOutput',
+            doc: {
+                summary: 'This mutation authorizes the user by token, after confirming the phone number.\n\n' +
+                    'If the user is not registered, then he will be created with the data that is passed in the payload (user data).\n\n' +
+                    'If the existing user is missing some fields, then these fields will be taken from the payload (user data) and updated.\n\n' +
+                    'This mutation is not available for service users!',
+                errors: Object.fromEntries(Object.entries(ERRORS).filter(([, error]) => typeof error === 'object')),
```

It's better to change the ERRORS to include objects. Really you hide real error by this filter